### PR TITLE
add datatables-select w/ npm auto-update

### LINKED ISSUE #8515: [Request] Add datatables select extension
datatables select extension
https://github.com/DataTables/Select
MIT License
https://datatables.net/extensions/select/

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/35998935-request-add-datatables-select-extension?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #8515 
@pvnr0082t please help me review this pull request, thank you.
Manually add v1.0.0 from website http://cdn.datatables.net/select/1.0.0/ and css file from other git repo https://github.com/DataTables/Dist-DataTables-Select-DataTables
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/DataTables/Select
- Official website (optional, not the repository): https://datatables.net/extensions/select/
- NPM package url (optional): https://www.npmjs.com/package/datatables.net-select
- GitHub / Bitbucket popularity (required):
  - Count of stars: 10
  - Count of watchers: 4
  - Count of forks: 12
- NPM download stats (optional):
  - Downloads in the last day: 64
  - Downloads in the last week: 233
  - Downloads in the last month: 1059
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR
- [ ] The lib has notable popularity
  - [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry
- [ ] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

close #8515, cc @DataTables

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
Test failed @kennynaoh

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have fixed that problem.
Please help me review it again.
Thanks!

**Comment by PeterDaveHello** [comment]
@kennynaoh any reason to not use its npm package - `datatables.net-select` ?

**Comment by PeterDaveHello** [comment]
Looks like we also need the css file(s), though there is no pre-built css in repo.

**Comment by kennynaoh** [comment]
@PeterDaveHello 
So I need to suggest author to provide css files in this situation?

**Comment by kennynaoh** [comment]
Because I found that it is not a same git repo, right?
https://github.com/DataTables/Dist-DataTables-Select

**Comment by PeterDaveHello** [comment]
Just a distribution purpose repo ... 
> github.com/DataTables/**Dist**-DataTables-Select

**Comment by kennynaoh** [comment]
Sorry, I don't know what you mean.
You mean it should use npm auto-update?
Thanks!

**Comment by PeterDaveHello** [comment]
That's an official npm package, we should use npm auto-update here.

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have changed to npm auto-update.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
> Looks like we also need the css file(s)
So why is there no any css file in your commit? @kennynaoh

**Comment by kennynaoh** [comment]
@PeterDaveHello 
So I need to suggest the author to provide the css files?

**Comment by PeterDaveHello** [comment]
You should check what the problem is.

**Comment by kennynaoh** [comment]
@PeterDaveHello 
Is this a problem?
`we also need the css file, but there is no pre-built css in repo.`
So what should I do?
Thanks!

**Comment by PeterDaveHello** [comment]
@kennynaoh Oh it is a problem, can you discuss with the author to confirm if css files can be provided?  Thanks.

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 
I have added the css file.
Please help me review it again.
Thanks!

**Comment by kennynaoh** [comment]
I think it's better to grab from git repo, too. Thanks!
@pvnr0082t , @x09326 
Please help me review it again.
Thanks!

**Comment by x09326** [comment]
@kennynaoh 
There is still a css file with `.map` in v1.1.0.

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 
fixed it.
Please help me review it again.
Thanks!

**Comment by x09326** [comment]
LGTM

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
Is there v1.0.0 in repo https://github.com/DataTables/Dist-DataTables-Select-DataTable ? I didn't see that.

What the reason why we need to get files from 2 sources? How do we know that's correct?

**Again**, you should tell us before been asked. @pvnr0082t @x09326 don't you have the same question?

**Comment by kennynaoh** [comment]
>Is there v1.0.0 in repo https://github.com/DataTables/Dist-DataTables-Select-DataTables ? I didn't see that.

No, the author offer the website  http://cdn.datatables.net/select/1.0.0/ about the version 1.0.0.

>What the reason why we need to get files from 2 sources? How do we know that's correct?

Because the compiled css files are from another repo https://github.com/DataTables/Dist-DataTables-Select-DataTables which is offered by author.

**Comment by PeterDaveHello** [comment]
So how do we know it will come from anther repo? That's the concern/question but you didn't answer. No any reference?

**Comment by kennynaoh** [comment]
OK, I will ask author.

**Comment by PeterDaveHello** [comment]
I'm not asking you to ask the author, but just give some reference for the non-common cases, or how do we know?

**Comment by kennynaoh** [comment]
you mean the issue which I discuss with author?
https://github.com/DataTables/Select/issues/26
Thanks!

**Comment by PeterDaveHello** [comment]
@kennynaoh I know there is DataTables/Select#26, but could you just tell us what's going on there, and maybe the summary, especially this issue is not closed yet.

**Comment by kennynaoh** [comment]
@PeterDaveHello , sure
The summary is that this repo https://github.com/DataTables/Dist-DataTables-Select-DataTables is for the compiled file, and this repo https://github.com/DataTables/Select is a source repo. So I grab the compiled file from https://github.com/DataTables/Dist-DataTables-Select-DataTables
reference -> https://github.com/DataTables/Select/issues/26#issuecomment-259327755

There is not exist `v1.0.0` on compiled repo, so I grab the file from the website which is offered from author http://cdn.datatables.net/select/1.0.0/
reference -> https://github.com/DataTables/Select/issues/26#issuecomment-259389791

**Comment by PeterDaveHello** [comment]
Let's rebase this branch with master branch and then we could merge this, thanks.

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 
I have rebased to the latest master branch
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by x09326** [comment]
@kennynaoh please rebase to the latest master branch, thanks.

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 I have rebased to the latest master branch. Please help me review it again. Thanks!

**Comment by x09326** [comment]
LGTM
ping @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@kennynaoh can you double check if the author can provide them in a single package or repository? The auto-update config can't update from two sources.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

File: ajax/libs/datatables.net-select/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "homepage": "https://datatables.net/extensions/select/",
+  "license": "MIT",
+  "dependencies": {
+    "jquery": ">=1.7.0",
+    "datatables": ">=1.10.8"
+  },
+  "npmName": "datatables.net-select",
+  "npmFileMap": [
+    {
+      "basePath": "",
```
Comment (xtuc) [inline ajax/libs/datatables.net-select/package.json]: use js as basePath

**Comment by sashberd** [comment]
moved to #13708

**Pull request closed without merging**